common/containers/Tabs/SendTransaction/components/ConfirmationModal.jsx

```diff
@@ -56,24 +61,52 @@ export default class ConfirmationModal extends React.Component {
       }
     }, 1000);
 
-    this.props.wallet.getAddress().then(address => {
-      this.setState({ address });
-    });
+    this._setWalletAddress(this.props.wallet);
   }
 
   componentWillUnmount() {
     clearTimeout(this.readTimer);
   }
 
+  _setWalletAddress(wallet: BaseWallet) {
+    wallet.getAddress().then(fromAddress => {
```

Comment:
I think I actually made this suggestion @wbobeirne, so no need to take the blame :)

However, I do agree that it makes more sense to construct the wallet instance with the address since we only need to ask once at unlock, and never again (unlike sign).